packages/account-web-vault/src/web-user.ts

```diff
@@ -0,0 +1,250 @@
+import { Network, Context, Profile } from '@verida/client-ts';
+import { VaultAccount, hasSession, VaultAccountConfig } from './index';
+import { EventEmitter } from 'events'
+import { MessageSendConfig } from '@verida/client-ts/dist/context/interfaces';
+
+const VeridaEvents = new EventEmitter()
+
+export interface WebUserProfile {
+    name?: string
+    avatarUri: string
+    country?: string
+    description?: string
+}
+
+export interface WebUserConfig {
+    accountConfig: VaultAccountConfig
+    clientConfig: ClientConfig
+    contextConfig: ContextConfig
+    networkConfig: NetworkConnectionConfig
+}
+
+export interface WebUserMessage {
+    subject: string
+    text: string
+    link?: WebUserMessageLink
+}
+
+export interface WebUserMessageLink {
+    url: string
+    text: string
+}
+
+/**
+ * 
+ * @event profileChanged
+ * @event connected
+ * @event disconnected
+ */
+export class WebUser {
+
+    private static config: WebUserConfig
+    private static context?: Context
+    private static account?: VaultAccount
+    private static profile?: WebUserProfile
+    private static did?: string
+
+    private static connecting: Promise<boolean>
+    private static profileConnection: Profile
+
+    public static configure(config: WebUserConfig) {
+        WebUser.config = config
+    }
+
+    public static async getContext() {
+        await WebUser.requireConnection()
+        return WebUser.context
+    }
+
+    public static async getAccount() {
+        await WebUser.requireConnection()
+
+        return WebUser.account
+    }
+
+    public static async getDid() {
+        await WebUser.requireConnection()
+
+        return WebUser.did
+    }
+
+    public static async getPublicProfile(force: boolean = false) {
```

Comment:
Worth adding jsdoc mentionning the arg `force` is to refetch the profile instead of potentially using the cache

Reply:
Fixed. Renamed to `ignoreCache` to be clearer.